gulpfile.js

```diff
@@ -0,0 +1,122 @@
+const fs = require('fs');
+const path = require('path');
+const gulp = require('gulp');
+const replace = require('gulp-replace');
+const rename = require('gulp-rename');
+const clean = require('gulp-clean');
+const readlineSync = require('readline-sync');
+const {execSync} = require('child_process');
+
+const E2E_PROJECT_NAME = 'fabric8-test';
+const E2E_REPO = 'https://github.com/fabric8io/fabric8-test.git';
+
+const E2E_PROJECT = path.resolve(__dirname, '../', E2E_PROJECT_NAME);
+const TEST_DIR = path.resolve(E2E_PROJECT, 'ee_tests');
+
+const CONFIG_DIR = path.resolve(TEST_DIR, 'config');
+const CONFIG_NAME = 'local_osio.conf.sh';
+const CONFIG = path.resolve(CONFIG_DIR, CONFIG_NAME);
+const CONFIG_TEMPLATE_NAME = 'local_osio.conf.sh.template';
+const CONFIG_TEMPLATE = path.resolve(CONFIG_DIR, CONFIG_TEMPLATE_NAME);
+
+gulp.task('e2e:clean', () => {
+  if (fs.existsSync(E2E_PROJECT)) {
+    console.log(`Deleting project ${E2E_PROJECT}`);
+    return gulp.src(E2E_PROJECT, {read: false}).pipe(clean({force: true}));
+  }
+  return Promise.resolve();
+});
+
+// fetch and update e2e tests
+gulp.task('e2e:fetch', async () => {
```

Comment:
`gulp.task` is the old way. In v4 they moved to `function`. 

https://gulpjs.com/docs/en/getting-started/creating-tasks

Reply:
Updated usage to align with v4 syntax.